### ISSUE TITLE
FR: Export audio file?

### ISSUE THREAD

**Issue by randomuser112358** [tube42/drumon#13]
Hello.

First of all thank you for this nice app.

Would it be possible to add audio (MIDI/OGG/other) file export? Would be nice for ringtones.

**Comment by tube42** [comment]
Its pretty straight forward to export to WAV. The only problem is that it would require storage access permission which some people (including me) don't like.

**Comment by francwalter** [comment]
Could you give a hint how to proceed?
Maybe as a plugin, so everybody can choose...
Would be a great feature.

**Comment by catprisbrey** [comment]
Indeed, when I use the export, it copies to the clipboard and the notification says it can be emailed from there, but it appears to just be short text output, but I'm uncertain where to go from here.

**Comment by tube42** [comment]
import/export on the save page does not generate a wave file, it creates a textual representation of the current track so you can send your latest masterpiece to your friend via email. 

sorry for the confusion :)

What you need is to render-to-file, but that would require the app to access your phones storage. That is currently not supported because I don't want my app to have access to your files. 

I think there is a solution to this in Android 10, but I need to investigate it further.

**Comment by SynapseScribe** [comment]
As long as the app does not have internet connection, and it's open source, which is the problem of having storage access ? Even if i like the app, i find it a bit useless in the end without being able to export my masterpiece as mp3 or wav or whatever.

**Comment by ameaninglessname** [comment]
Why just generate the file at the app's data folder?
Then, it's no need for storage access?

**Comment by tube42** [comment]
You cannot access the local app folder from other apps or a file browser.

I have added initial code to use the system file picker which bypasses the access requirements but I have not had time to code the rest.

**Comment by orogenic** [comment]
Is this a permissions concern if the permission can be denied by default and the user must explicitly grant it in order to save the rendering to storage? Is storage access required to share a rendering to another app? Or is storage access required to render to memory then share?
